lib/datadog.ts

```diff
@@ -8,36 +8,88 @@
 
 import * as cdk from "@aws-cdk/core";
 import * as lambda from "@aws-cdk/aws-lambda";
-import { applyLayers, redirectHandlers, addForwarder } from "./index";
+import { applyLayers, redirectHandlers, addForwarder, applyEnvVariables, defaultEnvVar } from "./index";
 
 export interface DatadogProps {
-  lambdaFunctions: lambda.Function[];
   pythonLayerVersion?: number;
   nodeLayerVersion?: number;
   addLayers?: boolean;
   forwarderARN?: string;
+  flushMetricsToLogs?: boolean;
+  site?: string;
+  apiKey?: string;
+  apiKMSKey?: string;
+  enableDDTracing?: boolean;
+  injectLogContext?: boolean;
 }
 
+export const site_list: string[] = [
+  "datadoghq.com",
+  "datadoghq.eu",
+  "us3.datadoghq.com",
+  "ddog-gov.com"
+]
+
 export class Datadog extends cdk.Construct {
-  /** allows accessing the counter function */
+  scope: cdk.Construct;
+  props: DatadogProps
   constructor(scope: cdk.Construct, id: string, props: DatadogProps) {
     super(scope, id);
-    if (props.addLayers === undefined) {
-      props.addLayers = true;
+    this.scope = scope;
+    this.props = props;
+  }
+
+  public addLambdaFunctions(lambdaFunctions: lambda.Function[]){
+    if (this.props.addLayers === undefined) {
+      this.props.addLayers = defaultEnvVar.addLayers;
+    }
+    if (this.props.flushMetricsToLogs === undefined) {
+      this.props.flushMetricsToLogs = defaultEnvVar.flushMetricsToLogs;
+    }
+    if (this.props.flushMetricsToLogs === false && this.props.site === undefined) {
+      throw new Error('A site is required if flushMetricsToLogs is set to false.');
+    }
+    if (this.props.site === undefined) {
+      this.props.site = defaultEnvVar.site;
+    }
+    if (!(site_list.includes(this.props.site))) {
```

Comment:
You might want to `toLower` `this.props.site` as a convenience thing for future maintainers/users.